tests/test_core_perim.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from geomeppy.geom.core_perim import core_perim_zone_coordinates
+
+footprint_1 = [(-20, 20), (30, -30), (50, 0), (50, 65), (-25, 50)]
+expected_footprint_1 = {'Perimeter_Zone_5':
+                        [(-19.27212384434131, 46.04655571753895),
+                         (-25.0, 50.0),
+                         (-20.0, 20.0),
+                         (-15.331451032074877, 22.402518843940353)],
+                        'Perimeter_Zone_4':
+                        [(45.0, 58.900980486407214),
+                         (50.0, 65.0),
+                         (-25.0, 50.0),
+                         (-19.27212384434131, 46.04655571753895)],
+                        'Perimeter_Zone_3':
+                        [(45.0, 1.5138781886599777),
+                         (50.0, 0.0),
+                         (50.0, 65.0),
+                         (45.0, 58.900980486407214)],
+                        'Perimeter_Zone_2':
+                        [(29.2228758492822, -22.15180803741673),
+                         (30.0, -30.0),
+                         (50.0, 0.0),
+                         (45.0, 1.5138781886599777)],
+                        'Perimeter_Zone_1':
+                        [(-15.331451032074877, 22.402518843940353),
+                         (-20.0, 20.0),
+                         (30.0, -30.0),
+                         (29.2228758492822, -22.15180803741673)],
+                        'Core_Zone':
+                        [(-15.331451032074877, 22.402518843940353),
+                         (-19.27212384434131, 46.04655571753895),
+                         (45.0, 58.900980486407214),
+                         (45.0, 1.5138781886599777),
+                         (29.2228758492822, -22.15180803741673),
+                         (-15.331451032074877, 22.402518843940353)]}
+
+
+footprint_2 = [(0, 0), (30, 0), (30, 20), (0, 20)]
+expected_footprint_2 = {'Core_Zone': [(5.0, 5.0), (5.0, 15.0),
+                                      (25.0, 15.0), (25.0, 5.0), (5.0, 5.0)],
+                        'Perimeter_Zone_4': [(5.0, 15.0), (0.0, 20.0),
+                                             (0.0, 0.0), (5.0, 5.0)],
+                        'Perimeter_Zone_3': [(25.0, 15.0), (30.0, 20.0),
+                                             (0.0, 20.0), (5.0, 15.0)],
+                        'Perimeter_Zone_2': [(25.0, 5.0), (30.0, 0.0),
+                                             (30.0, 20.0), (25.0, 15.0)],
+                        'Perimeter_Zone_1': [(5.0, 5.0), (0.0, 0.0),
+                                             (30.0, 0.0), (25.0, 5.0)]}
+
+
+def test_core_perim():
```

Comment:
While we're looking at tests, a few points:

 1. this would probably be better as two separate tests. One for testing the footprints and expected results, and one for the error.
 
 2. What are you trying to test with the two footprints? Are they different in a meaningful way?
 
 3. `NotImplementedError` isn't the expected error. We want to be sure that the `ValueError` gets raised with our own message, "Perimeter depth is too great"

4. Why doesn't `get_core_area` get called in any tests? And why does it return a bool? https://codecov.io/gh/jamiebull1/geomeppy/src/master/geomeppy/geom/core_perim.py

Reply:
1. Sounds good.
2. Yes and no. It matters less now because we're using `shapely`'s `buffer`, it was more meant for what I had before.
3. Ok, I'll change that.
4. I'll create a test where it gets called.

Reply:
On 4, the name doesn't sound like what it appears to be doing. If it is actually returning the area, can you add a docstring to clarify how it's doing it and why it needs to be done like this rather than using `Polygon.area`. Otherwise, can you rename it - or remove it if it's not used.

Reply:
Will do!